buildconfig/Jenkins/cppcheck.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -ex
+
+if [[ ${JOB_NAME} == *pull_requests* ]]; then
+    # This relies on the fact pull requests use pull/$PR-NAME
+    # which squashes the branch into a single merge commit
+    cd $WORKSPACE
```

Comment:
Is it possible to use the [check_for_changes](https://github.com/mantidproject/mantid/blob/master/buildconfig/Jenkins/check_for_changes) script at all?

Reply:
The only downside is we will have to change from a shallow clone depth of 2 to a full clone, as we use `HEAD^2` rather than HEAD~2

Let's try it out and see how long the clone takes

Reply:
I seem to be running into problems with `git merge-base HEAD^2 HEAD^1` not having a common ancestor?

I can reproduce this locally too, `git test-pr 29456 && git merge-base HEAD^2 HEAD^1` returns nothing. But I can't see why this works for unit test build hence my confusion